File: kotlin/sprint2/B/Solution.kt

import kotlin.jvm.internal.Reflection.function


/**
 * Comment it before submitting
 *
 * class Node<V>(var value: V, var next: Node<V>?) {}
 */

fun solution(head: Node<String>?) {
    // Your code
    // ヽ(´▽`)/
}

fun test() {
    val node3 = Node("node3")
    val node2 = Node("node2", node3)
    val node1 = Node("node1", node2)
    val node0 = Node("node0", node1)
    solution(node0)
    /*
    Output is:
    node0
    node1
    node2
    node3
    */
}

File: swift/sprint2/B/solution.swift

/**
Comment it before submitting

open class Node {
   
   public var value: String
   public var next: Node?
   
   public init(_ value: String) {
     self.value = value
     self.next = nil
   }
 }
 **/

func solution(head: Node?) {
    // Your code
    // ヽ(´▽`)/
}


func test() {
    let node3 = Node("node3")
    let node2 = Node("node2")
    node2.next = node3
    let node1 = Node("node1")
    node1.next = node2
    let node0 = Node("node0")
    node0.next = node1
    solution(head: node0)
    /*
    Output is:
    node0
    node1
    node2
    node3
    */
}


